pkg/apis/pingcap/v1alpha1/tidbcluster_component.go

```diff
@@ -247,8 +272,73 @@ func (a *componentAccessorImpl) TerminationGracePeriodSeconds() *int64 {
 	return a.ComponentSpec.TerminationGracePeriodSeconds
 }
 
-func buildTidbClusterComponentAccessor(spec *TidbClusterSpec, componentSpec *ComponentSpec) ComponentAccessor {
+func (a *componentAccessorImpl) TopologySpreadConstraints() []corev1.TopologySpreadConstraint {
+    tscs := a.topologySpreadConstraints
+	if a.ComponentSpec != nil && len(a.ComponentSpec.TopologySpreadConstraints) > 0 {
+		tscs = a.ComponentSpec.TopologySpreadConstraints
+	}
+
+	if len(tscs) == 0 {
+	    return nil
+	}
+
+	ptscs := make([]corev1.TopologySpreadConstraint, 0, len(tscs))
+	for _, tsc := range tscs {
+		ptsc := corev1.TopologySpreadConstraint{
+			MaxSkew:           1,
+			TopologyKey:       tsc.TopologyKey,
+			WhenUnsatisfiable: corev1.DoNotSchedule,
+		}
+		componentLabelVal := getComponentLabelValue(a.component)
+		var l label.Label
+		switch a.kind {
+		case TiDBClusterKind:
+			l = label.New()
+		case DMClusterKind:
+			l = label.NewDM()
+		}
+		l[label.ComponentLabelKey] = componentLabelVal
+		l[label.InstanceLabelKey] = a.name
+		ptsc.LabelSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string(l),
```

Comment:
It seems that we cannot differentiate the TidbClusters with the same name in the different namespaces, right?

Reply:
Yes, the label selector is namespaced scope.

Reply:
OK, please make it clear in the comments of the API doc and the user doc that if users use this feature, they have to make sure that they don't use the same TidbCluster name.